front/components/app/blocks/DataSource.jsx

```diff
@@ -126,6 +126,12 @@ export default function DataSource({
     onBlockUpdate(b);
   };
 
+  const handleFullTextChange = (full_text) => {
```

Comment:
```suggestion
  const handleFullTextChange = useCallback(full_text) => {
```
Here you could use a useCallback() to avoid some useless re-render, but it is not a big deal.

That being said, I found that we have disabled some react linter rules that are quite important, especially the `"react-hooks/exhaustive-deps": 0,` in `/front/.eslintrc.json`, which masks some useEffect() dependencies error like this one:
<img width="1176" alt="Screen Shot 2023-03-29 at 17 16 49" src="https://user-images.githubusercontent.com/358965/228586341-120d9463-f804-44f0-87d8-37dcb63bb784.png">

We should probably get those warnings back, except if I missed something.